### PR TITLE
feat: introduce devnetVersion for devnet breaking changes; always use DGW on devnets

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
This adds a devnetVersion to both the devnet coinbase, and to the uacomment. This allows us to introduce breaking changes to devnets that are not silent to end users. Previously we introduced a breaking change that then caused issues later because we forgot about it. Now that I want to do another breaking change, it probably makes sense to introduce this logic now.

The second bit, is a breaking change that removes the bitcoin difficulty algorithm from devnets, this is a breaking change for devnets, additionally, since this PR changes the uacomment, it may be a breaking change from the perspective of mobile clients.

fixes https://github.com/dashpay/dash/issues/4292

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
I must be missing smth but why using devnet names like `devnet_1`, `devnet_2` etc. is not enough? Also, using different devnet versions for the same devnet name will cause conflicts I think (cause datadir would be the same for both devnets in such cases).

EDIT: de681538d4bd9707cde06f87778173b027df02a3 is ok assuming new binaries are going to be used on new devnets only

**Comment by PastaPastaPasta** [comment]
Assume we deploy a devnet, devnet-malort for example. Then we merge some breaking change. Now any nodes running the new software CAN connect to devnet-malort, and maybe could sync the entire chain etc, until the breaking change happens and we get some cryptic seg fault on these nodes running the new code / stall etc. If you have this devnet version, then 1. they can't sync past genesis because it'll not match, and 2. they can't connect to peers because ua agent is wrong, this makes it clear there is a compatibility issue, and prevents you wasting time trying to connect to the devnet and debug why it's crashing. Does that make sense?

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
Hmm, ok, this makes sense I guess. `p2p_connect_to_devnet.py` fails, a23b92be0835e9a6038a377cb342af983772c979 should help.

**Review by UdjinM6 — APPROVED** [review]
utACK

File: src/chainparams.cpp [hunk 1]
```diff
@@ -682,7 +682,8 @@ class CDevNetParams : public CChainParams {
         assert(consensus.hashGenesisBlock == uint256S("0x000008ca1832a4baf228eb1553c03d3a2c8e02399550dd6ea8d65cec3ef23d2e"));
         assert(genesis.hashMerkleRoot == uint256S("0xe0028eb9648db56b1ac77cf090b99048a8007e2bb64b68f092c03c7f56a662c7"));
 
-        devnetGenesis = FindDevNetGenesisBlock(genesis, 50 * COIN);
+        devnetVersion = 1; // This should be bumped every time there is some breaking change to devnets
+        devnetGenesis = FindDevNetGenesisBlock(genesis, /*reward=*/50 * COIN, /*devnet_version=*/devnetVersion);
```
Comment (HashEngineering) [inline src/chainparams.cpp:686]: Doesn't setting this to 1 mean that all devnets will be on version 1?

Could one devnet be set to 2 and another be set to 3 in the config, but here, that version is reset to 1?

**Pull request merged**